plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -45,12 +45,15 @@ private AdvancedSectionExpansionHandler getExpansionHandler() {
 
       @Override
       public void handleExpansionEvent(ExpansionEvent e) {
-        Composite parent = ((Control)e.getSource()).getParent();
-        parent.layout();
+        Composite expandableComposite = (Composite)e.getSource();
+        DeployPreferencesPanel panel = (DeployPreferencesPanel) expandableComposite.getParent();
```

Comment:
Since the `DeployPropertyPage` creates the panel, why don't you stash it in a field.